libkbfs/rekey_fsm.go

```diff
@@ -435,25 +456,23 @@ func (m *rekeyFSM) loop() {
 
 // Event implements RekeyFSM interface for rekeyFSM.
 func (m *rekeyFSM) Event(event RekeyEvent) {
-	select {
-	case m.reqs <- event:
-		return
-	default:
-	}
-	go func() {
-		// We use a spinning loop here to avoid doing a naked `m.reqs <- event`,
-		// which in racy conditions could cause a panic due to sending to closed
-		// channel. The spinning loop is fine here since reactToEvent
+	for {
+		// We use a spinning loop here to avoid doing a naked `m.reqs <-
+		// event`, which in racy conditions could cause a panic due to sending
+		// to closed channel. The spinning loop is fine here since reactToEvent
```

Comment:
Leaving the read loop will leak a goroutine, but leaving a channel open won't. Plus, using channels & channels for synchronization is clean because they can go in the same `select`. Mutexes can't.

In general, if you're guarding channels with mutexes, there's probably something wrong.